### PR TITLE
Add job retries support to AzureDevOpsWorker

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

File: src/Microsoft.Crank.AzureDevOpsWorker/Program.cs [hunk 1]
```diff
@@ -160,57 +160,70 @@ private static async Task MessageHandler(ProcessMessageEventArgs args)
...
-                            Console.WriteLine($"{LogNow} Job is completed ({record.Result}), interrupting...");
+                            if (record != null && record?.State == "completed")
+                            {
+                                Console.WriteLine($"{LogNow} Job is completed ({record.Result}), interrupting...");
+
+                                driverJob.Stop();
+                            }
 
-                            driverJob.Stop();
+                            await Task.Delay(TaskLogFeedDelay);
```
Comment (BrennanConroy) [inline src/Microsoft.Crank.AzureDevOpsWorker/Program.cs:221]: Should this be in an `else`? No need to delay if we stopped the driver right?

File: src/Microsoft.Crank.AzureDevOpsWorker/Program.cs [hunk 2]
```diff
@@ -160,57 +160,70 @@ private static async Task MessageHandler(ProcessMessageEventArgs args)
...
-                                driverJob.Stop();
-                            }                                
-                        }
+                                if (!success)
+                                {
+                                    Console.ForegroundColor = ConsoleColor.DarkYellow;
+                                    Console.WriteLine($"{LogNow} SendTaskLogFeedsAsync failed. If the task was canceled, this jobs should be stopped.");
+                                    Console.ResetColor();
+
+                                    driverJob.Stop();
```
Comment (BrennanConroy) [inline src/Microsoft.Crank.AzureDevOpsWorker/Program.cs:202]: `continue;`?

File: src/Microsoft.Crank.AzureDevOpsWorker/Program.cs (hunk 1 above)
Comment (sebastienros) [reply src/Microsoft.Crank.AzureDevOpsWorker/Program.cs]: Better

**Review by BrennanConroy — APPROVED** [review]

File: src/Microsoft.Crank.AzureDevOpsWorker/Program.cs (hunk 2 above)
Comment (sebastienros) [reply src/Microsoft.Crank.AzureDevOpsWorker/Program.cs]: I see the point, so I removed this line and let the next part do it. But I used this in the timeout detection instead

**Pull request merged**